contracts/src/token/erc20/extensions/flashmint.rs

```diff
@@ -0,0 +1,164 @@
+
+    fn flash_fee(&self, token:Address, amount:U256)  -> Result<U256, Self::Error> ;
+
+    fn flash_loan(
+        &mut self,
+        receiver: Address,
+        token:Address,
+        amount:U256,
+        data:Bytes
+    ) -> Result<bool, Self::Error>;
+}
+
+/// A Permit error.
+#[derive(SolidityError, Debug)]
+pub enum Error {
+    UnsupportedToken(ERC3156UnsupportedToken),
+    ExceededMaxLoan(ERC3156ExceededMaxLoan),
+    InvalidReceiver(ERC3156InvalidReceiver),
+
+    /// Error type from [`Erc20`] contract [`erc20::Error`].
+    Erc20(erc20::Error),
+
+}
+
+const RETURN_VALUE: B256 =
+    b256!("439148f0bbc682ca079e46d6e2c2f0c1e3b820f1a291b069d8882abf8cf18dd9");
+
+impl IERC3156FlashLender for Erc20 {
+    
+    type Error = Error ;
+
+    /// Returns the maximum amount of tokens that can be borrowed
+    /// from this contract in a flash loan.
+    ///
+    /// For tokens that are not supported, this function returns
+    /// `U256::MIN`.
+    ///
+    /// * `token` - The address of the ERC-20 token that will be loaned.
+    fn max_flash_loan(&self, token:Address) -> U256 {
+        if token == contract::address() {
+            return   U256::MAX - self.total_supply();
+        }
+        U256::MIN
+    }
+
+    /// Calculates the fee for a flash loan.
+    ///
+    /// The fee is a fixed percentage of the borrowed amount.
+    ///
+    /// If the token is not supported, the function returns an
+    /// `UnsupportedToken` error.
+    ///
+    /// * `token` - The address of the ERC-20 token that will be loaned.
+    /// * `amount` - The amount of tokens that will be loaned.
+    fn flash_fee(&self,token:Address, amount:U256)  -> Result<U256, Error> {
+        if token != contract::address() {
+            return Err(Error::UnsupportedToken(ERC3156UnsupportedToken { token}));
+        }
+        Ok(self._flash_fee(token , amount))
+    }
+
+    /// Executes a flash loan.
+    ///
+    /// This function is a part of the ERC-3156 (Flash Loans) standard.
+    ///
+    /// * `receiver` - The contract that will receive the flash loan.
+    /// * `token` - The ERC-20 token that will be loaned.
+    /// * `amount` - The amount of tokens that will be loaned.
+    /// * `data` - Arbitrary data that can be passed to the receiver contract.
+    ///
+    /// The function must return `true` if the flash loan was successful,
+    /// and revert otherwise.
+    fn flash_loan(
+        &mut self,
+        receiver: Address,
+        token:Address,
+        value:U256,
+        data:alloy_primitives::Bytes
+    ) -> Result<bool, Error> {
+        let  max_loan =  self.max_flash_loan(token);
+        if value > max_loan {
+            return Err(Error::ExceededMaxLoan(ERC3156ExceededMaxLoan { maxLoan: max_loan }));
+        }
+
+        let fee =  self.flash_fee(token, value)?;
+        self._mint(receiver, value)?;
+        let loan_reciver = IERC3156FlashBorrower::new(receiver);
+        let config = Call::new();
```

Comment:
```suggestion
        let call = Call::new();
```
nit: this is the name convention for `Call` instances